pyScienceMode2/Channel.py

```diff
@@ -150,3 +315,124 @@ def get_enable_low_frequency(self) -> bool:
         Returns enable_low_frequency.
         """
         return self._enable_low_frequency
+
+    def set_frequency(self, frequency: float):
+        """
+        Set the frequency for a channel
+
+        Parameters
+        ----------
+        frequency: float
+            Frequency of the channel (Hz)
+        """
+        if frequency <= 0:
+            raise ValueError("frequency must be positive.")
+        self._period = 1000.0 / frequency
+
+        self.generate_pulse()
+
+    def get_frequency(self) -> float:
+        """
+        Returns the frequency of a channel
+        """
+        return 1000.0/self._period
+
+    def get_ramp(self):
+        """
+        Returns the ramp of a channel
+        """
+        return self._ramp
+
+    def set_ramp(self, ramp: int):
+        """
+        Set the ramp for a channel
+
+        Parameters
+        ----------
+        ramp: int
+            Ramp of the channel (pulses)
+        """
+        self._ramp = ramp
+        self.check_value_param()
+        self.generate_pulse()
+
+    def add_point(self, pulse_width: float, amplitude: float):
+        """
+        Add a point to the list of points for a channel. One channel can pilot 16 points.
+
+        Parameters
+        ----------
+        pulse_width: float
+            Width of the stimulation. [0,4093] μs
+        amplitude: float
+            Current to send in the channel. [-150,150] milli amp
+
+        Returns
+        -------
+        point: Point
+        """
+        if len(self.list_point) < Channel.MAX_POINTS:
+            point = Point(pulse_width, amplitude)
+            self.list_point.append(point)
+        else:
+            raise ValueError(f"Cannot add more than {Channel.MAX_POINTS} points to a channel")
+        return point
+
+    def generate_pulse(self):
+        """
+        Generate a pulse for a channel. The pulse is generated according to the mode and the parameters given.
+        """
+        if self.device_type == "RehastimP24":
+            if self._mode == Channel.MODE["Single"]:  # Create a biphasic pulse for the channel
+                self.create_biphasic_pulse(self._amplitude, self._pulse_width)
+            elif self._mode == Channel.MODE["Doublet"]:  # Create a doublet pulse for the channel
+                self.create_doublet(self._amplitude, self._pulse_width)
+            elif self._mode == Channel.MODE["Triplet"]:  # Create a triplet pulse for the channel
+                self.create_triplet(self._amplitude, self._pulse_width)
+
+
+class Point:
```

Comment:
Flag to protect user from unsave stimulation shape
if negative shape is different than positive, raiseerror or enable safe_mode flag to true

Reply:
Done.